crates/c-api/include/wasmtime/store.h

```diff
@@ -243,6 +243,30 @@ WASM_API_EXTERN void wasmtime_context_set_epoch_deadline(wasmtime_context_t *con
  */
 WASM_API_EXTERN void wasmtime_store_epoch_deadline_callback(wasmtime_store_t *store, wasmtime_error_t* (*func)(wasmtime_context_t*, void*, uint64_t*), void *data);
 
+/**
+ * \brief Insert a file descriptor into the underlying WASI context. 
+ * 
+ * This is used to dynamically bind a file descriptor from the host to a file 
+ * descriptor in an already-built WASI context to make it available to WASI APIs.
+ * 
+ * The `guest_fd` argument is the number of the file descriptor by which it will
+ * be known in WASM and the `host_fd` is the number of the file descriptor on
+ * the host.
+ */
+WASM_API_EXTERN void wasmtime_context_insert_file(wasmtime_context_t *context, uint32_t guest_fd, uint32_t host_fd, uint32_t access_mode);
+
+/**
+ * \brief Push a file descriptor into the underlying WASI context.
+ * 
+ * This is used to dynamically bind a file descriptor from the host to the next 
+ * available file descriptor in an already-built WASI context to make it available 
+ * to WASI APIs.
+ * 
+ * The `host_fd` argument is the number of the file descriptor on the host and the
+ * `access_mode` is the access mode of the file descriptor.
+ */
```

Comment:
Could this documentation, and the above block, mention the requirement to configure WASI first? If that's not done then this function will abort the process due to a failed assert.

Reply:
Sure, glad to make add this to the documentation.